awx/main/models/projects.py

```diff
@@ -418,6 +423,10 @@ def needs_update_on_launch(self):
                 return True
         return False
 
+    @property
+    def cache_id(self):
+        return str(self.last_job_id)
```

Comment:
This will never be a project `sync` job right?

Reply:
You're looking at the `Project` model definition. It is also defined for `ProjectUpdate` down below. If it's a normal sync, then it return's the project's last job (for core functionality). If it's a branch override case, we agreed to give up on cache invalidation, so it uses its own id, simply to make sure that it is never mistaken for the default branch's cache by other syncs.